src/core/dbus-manager.c

```diff
@@ -2092,11 +2092,18 @@ static int method_enable_unit_files_generic(
         if (r < 0)
                 return r;
 
-        r = sd_bus_message_read(message, "bb", &runtime, &force);
-        if (r < 0)
-                return r;
+        if (sd_bus_message_is_method_call(message, NULL, "EnableUnitFilesWithFlags")) {
+                r = sd_bus_message_read(message, "t", &flags);
+                if (r < 0)
+                        return r;
```

Comment:
let's refuse invalid flags here. i.e. all flags we don#t define at all, and those which we only want to use internally, i.e. extend the UnitFileFlags enum with a new value `_UNIT_FILE_FLAGS_MASK_PUBLIC = UNIT_FILE_RUNTIME|UNIT_FILE_PORTABLE|UNIT_FILE_FORCE`

And then use that mask to refuse flags that aren't in this mask here.

Reply:
done